app/code/Magento/Catalog/Block/Product/ImageFactory.php

```diff
@@ -76,6 +76,7 @@ public function __construct(
     private function getStringCustomAttributes(array $attributes): string
```

Comment:
I think the `array` restriction will result in the `TypeError` if the `$attributes` is `null`.
So I think the code ensuring `$attributes` is an array should be executed before calling this function